tooling/nargo_toml/src/lib.rs

```diff
@@ -518,7 +519,10 @@ pub fn resolve_workspace_from_toml(
     current_compiler_version: Option<String>,
 ) -> Result<Workspace, ManifestError> {
     let nargo_toml = read_toml(toml_path)?;
+    let lock = lock_git_deps().expect("Failed to lock git dependencies cache");
     let workspace = toml_to_workspace(nargo_toml, package_selection)?;
```

Comment:
It would be nice to add some docs to `toml_to_workspace` and the stuff it calls, because it's not obvious that it can result in git dependencies being cloned.

Reply:
I agree but I think that we should address the code issues here which are tightly coupling downloading and resolving of toml files.

Reply:
You mean we should restrict ourselves to just the code, as opposed to improving docs? Fair enough. 

I'm not sure we have addressed any tight coupling though.